client-sdk/go/types/transaction.go

```diff
@@ -22,6 +22,8 @@ type AuthProof struct {
 	Signature []byte `json:"signature,omitempty"`
 	// Multisig is for multisig authentication.
 	Multisig [][]byte `json:"multisig,omitempty"`
+	// Module is a flag to use module-controled decoding.
+	Module string `json:"module,omitempty"`
```

Comment:
I guess this doesn't differentiate between a module being set to an empty string and the module not even being there? For example if you use an empty string then the check below will pass. You could maybe use a `*string` instead and check that it is `nil`.

Reply:
the empty string case! :laughing: 

Go is weird in this, but it ends up not being that bad. we can require that the encoding scheme name be non-empty. Similarly, that `Signature []byte` up there requires that the signature not be zero length, which would be encoded as nil.

Reply:
ok I tried changing it to *string in a separate commit. although now you can't build AuthProofs in code as easily, with go not permitting `&"sceme-name"` address-of-constant antics

Reply:
Oh I see, yeah i would be fine also with changing it back and not allowing empty strings as long as it is consistent with what is done in the runtime.

Reply:
ok. adding a note about the non-empty requirement